modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -698,6 +704,39 @@ public SCIMResponse updateWithPATCH(String existingId, String scimObjectString,
                 } else {
                     throw new BadRequestException("Unknown operation.", ResponseCodeConstants.INVALID_SYNTAX);
                 }
+
+                List<String> scimAttributes = determineScimAttributes(operation);
+
+                for (String scimAttribute : scimAttributes) {
+                    String syncedAttribute = syncedAttributes.get(scimAttribute);
```

Comment:
Shall we add null check here.

Reply:
Fixed with 935345804f0a32308eb6175284e90a06f336f055